sphinx_immaterial/task_lists.py

```diff
@@ -0,0 +1,95 @@
+"""A custom directive that allows using checkboxes for lists"""
+from typing import List
+from docutils import nodes
+from docutils.parsers.rst import directives
+from sphinx.util.docutils import SphinxDirective
+from sphinx.application import Sphinx
+from sphinx.writers.html import HTMLTranslator
+
+
+def visit_checkbox_label(self: HTMLTranslator, node: nodes.Node):
+    attributes = {}
+    if node["custom"]:
+        attributes = {"class": "task-list-control"}
+    self.body.append(self.starttag(node, "label", **attributes))
+    self.body.append('<input type="checkbox"')
+    if node["disabled"]:
+        self.body.append(" disabled ")
+    if node["checked"]:
+        self.body.append(" checked ")
+    self.body.append(">")
+    if node["custom"]:
+        self.body.append('<span class="task-list-indicator"></span>')
+
+
+def depart_checkbox_label(self: HTMLTranslator, node: nodes.Node):
+    self.body.append("</label>")
+
+
+class checkboxLabel(nodes.container):
```

Comment:
docutils and sphinx use `snake_case` for node names --- best to be consistent here.